internal/controller/checkpointrestoreoperator_controller.go

```diff
@@ -47,18 +48,25 @@ import (
 
 type RetentionPolicy int
 
+// This constants are used to specify the policy criteria
+// for selecting which checkpoint archives to delete .
 const (
 	ByCount RetentionPolicy = iota
+	BySize
 )
 
 var (
 	GarbageCollector           garbageCollector
 	policyMutex                sync.RWMutex
 	checkpointDirectory        string = "/var/lib/kubelet/checkpoints"
 	quit                       chan bool
-	maxCheckpointsPerContainer int = 10
-	maxCheckpointsPerPod       int = 20
-	maxCheckpointsPerNamespace int = 30
+	maxCheckpointsPerContainer int               = 10
+	maxCheckpointsPerPod       int               = 20
+	maxCheckpointsPerNamespace int               = 30
+	maxCheckpointSize          resource.Quantity = resource.MustParse("1Ei")
+	maxTotalSizePerPod         resource.Quantity = resource.MustParse("1Ei")
+	maxTotalSizePerContainer   resource.Quantity = resource.MustParse("1Ei")
+	maxTotalSizePerNamespace   resource.Quantity = resource.MustParse("1Ei")
```

Comment:
I'm not sure if 1 exabyte is a very sane default value. We could potentially restrict this to something a little more reasonable, like 100 gigabytes or so? cc @rst0git